src/main/java/heronarts/lx/pattern/color/AlternatingPattern.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2013- Mark C. Slee, Heron Arts LLC
+ *
+ * This file is part of the LX Studio software library. By using
+ * LX, you agree to the terms of the LX Studio Software License
+ * and Distribution Agreement, available at: http://lx.studio/license
+ *
+ * Please note that the LX license is not open-source. The license
+ * allows for free, non-commercial use.
+ *
+ * HERON ARTS MAKES NO WARRANTY, EXPRESS, IMPLIED, STATUTORY, OR
+ * OTHERWISE, AND SPECIFICALLY DISCLAIMS ANY WARRANTY OF
+ * MERCHANTABILITY, NON-INFRINGEMENT, OR FITNESS FOR A PARTICULAR
+ * PURPOSE, WITH RESPECT TO THE SOFTWARE.
+ *
+ * @author Mike Schiraldi
+ */
+
+package heronarts.lx.pattern.color;
+
+import heronarts.lx.LX;
+import heronarts.lx.LXCategory;
+import heronarts.lx.color.LXColor;
+import heronarts.lx.color.LinkedColorParameter;
+import heronarts.lx.parameter.DiscreteParameter;
+import heronarts.lx.pattern.LXPattern;
+
+@LXCategory(LXCategory.COLOR)
+public class AlternatingPattern extends LXPattern {
+  public final LinkedColorParameter firstColor =
+          new LinkedColorParameter("First Color")
+                  .setDescription("First color to use");
+
+  public final LinkedColorParameter secondColor =
+          new LinkedColorParameter("Second Color")
+                  .setDescription("Second color to use");
+
+  public final DiscreteParameter stripeLength =
+          new DiscreteParameter("Length", 1, 10)
+                  .setDescription("Number of pixels to go before switching colors");
+
+  public AlternatingPattern(LX lx) {
+    // These colors will only be seen when the user switches modes:
+    this(lx, LXColor.WHITE, LXColor.BLUE);
+
+    this.firstColor.mode.setValue(LinkedColorParameter.Mode.PALETTE);
+    this.firstColor.index.setValue(1);
+
+    this.secondColor.mode.setValue(LinkedColorParameter.Mode.PALETTE);
+    this.secondColor.index.setValue(2);
+  }
+
+  public AlternatingPattern(LX lx, int firstColor, int secondColor) {
+    super(lx);
+    this.firstColor.setColor(firstColor);
+    this.secondColor.setColor(secondColor);
+    addParameter("firstColor", this.firstColor);
+    addParameter("secondColor", this.secondColor);
+    addParameter("stripeLength", this.stripeLength);
+  }
+
+  @Override
+  public void run(double deltaMs) {
+    LinkedColorParameter[] colorParameters = {this.firstColor, this.secondColor};
```

Comment:
This and the line below are going to allocate new dynamic arrays on every single iteration of the run loop that will need garbage collecting. Best to avoid that wherever possible. If you really want to use arrays here you should make them member variables on the class. But honestly, with just two colors here, I think the below is more readable. Also recommend pre-fetching a parameter value if it's going to be used many times... otherwise it's N function calls for a N-pixel project... the JVM can sort of optimize-it-away and CPU is plentiful these days, but best to avoid 50K+ function calls where you can.

```
int stripeLength = this.stripeLength.getValuei();
int color1 = this.firstColor.calcColor();
int color2 = this.secondColor.calcColor();
for (int i = 0; i < colors.length; ++i) {
    int phase = (i / this.stripeLength) % 2;
    this.colors[i] = (phase == 0) ? color1 : color2;
}
```

Reply:
Ah, I overestimated the sophistication of the optimizer. Good to know. Updated to use your code.

Reply:
It is something that I'm always curious about. It *might* be able to figure out those arrays are not-really-dynamic and reuse them, or at least they will just get smartly re-allocated when the next loop comes around asking for an array of the same size. I shouldn't speak definitively about this, as I don't really know how it works. But I try not to count on that sort of magic in cases where it is easily avoided, particularly inside the run() loops.